### DISCUSSION TITLE
How to get the current page

### DISCUSSION THREAD

**Discussion by lukavia** [sqlpage/SQLPage#110] [Q&A] [answered]
I'm trying to a navigation to my app. Going through the docs and examples it seams the most elegant solution is to create a table with the entries and then use tab component. 
```
select 'tab' as component;
select title, file as link from nav; 
```

However I would like the current page to be active so I want to add a column like `IF(sqlpage.url = file, true, false) as active`

My question is how to get the current page (sqlpage.url if you will) ?

Thank you

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !

This is not yet implemented, but I created an issue here: https://github.com/lovasoa/SQLpage/issues/114

For the time being, you'll just have to duplicate in each page

```sql
select title, file as link, file='/my_page_xxx.sql' as active from nav; 
```

*side note* : `IF(sqlpage.url = file, true, false)` can be expressed as simply `sqlpage.url = file`

**Comment by lukavia** [comment]
Thank you very much. 
In the meanwhile I've currently narrowed it down to one simple table: 

```sql
CREATE TABLE nav(page VARCHAR(20));
INSERT INTO nav VALUES('transactions'),('accounts'),('this month');
```

And then one can use: 

```sql
SELECT 'shell' as component, 'Transactions' as title, json_group_array(page) as menu_item from nav;
```

or

```sql
SELECT 'tab' as component;
SELECT page as title, page || '.sql' as link, (page = 'transactions') as active from nav; 
```

And then change only the string on each page.

**Comment by lovasoa** [comment, marked as answer]
I added [`sqlpage.path()`](https://sql.ophir.dev/functions.sql?function=path#function), it will be released in v0.15 :slightly_smiling_face:
